Source/WPE/Source/Input/LibinputServer.cpp

```diff
@@ -108,12 +113,33 @@ void LibinputServer::setPointerBounds(uint32_t width, uint32_t height)
     m_pointerBounds = { width, height };
 }
 
+void LibinputServer::configureHandleTouchEvents(bool enable, uint32_t touchWidth, uint32_t touchHeight)
+{
+    m_handleTouchEvents = enable;
+    if (enable)
+        m_touchSize = { touchWidth, touchHeight };
+    fprintf(stderr, "[LibinputServer] %s touch events.\n", enable ? "Enabling" : "Disabling");
+}
+
+
 void LibinputServer::processEvents()
 {
     libinput_dispatch(m_libinput);
 
     while (auto* event = libinput_get_event(m_libinput)) {
         switch (libinput_event_get_type(event)) {
+        case LIBINPUT_EVENT_TOUCH_DOWN: 
```

Comment:
These types should be tested for and handled in their numerical order. That means LIBINPUT_EVENT_TOUCH_* events are handled after the LIBINPUT_EVENT_POINTER_* events.